src/graphics/resolution.h

```diff
@@ -0,0 +1,81 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  A list common video resolutions.
+ */
+
+#ifndef GRAPHICS_RESOLUTION_H
+#define GRAPHICS_RESOLUTION_H
+
+namespace Graphics {
+
+struct Resolution {
+	int width;
+	int height;
+
+	Resolution(int w, int h) : width(w), height(h) { }
+};
+
+static std::vector<Resolution> initResolutions() {
+	// Add all standard resolutions to the list
+	std::vector<Resolution> resolutions;
+	resolutions.reserve(33);
+	resolutions.push_back(Resolution(7680, 4800));
+	resolutions.push_back(Resolution(7680, 4320));
+	resolutions.push_back(Resolution(6400, 4800));
+	resolutions.push_back(Resolution(6400, 4096));
+	resolutions.push_back(Resolution(5120, 4096));
+	resolutions.push_back(Resolution(5120, 3200));
+	resolutions.push_back(Resolution(4096, 3072));
+	resolutions.push_back(Resolution(4096, 1716));
+	resolutions.push_back(Resolution(3840, 2400));
+	resolutions.push_back(Resolution(3200, 2400));
+	resolutions.push_back(Resolution(3200, 2048));
+	resolutions.push_back(Resolution(2560, 2048));
+	resolutions.push_back(Resolution(2560, 1600));
+	resolutions.push_back(Resolution(2560, 1440));
+	resolutions.push_back(Resolution(2048, 1536));
+	resolutions.push_back(Resolution(2048, 1152));
+	resolutions.push_back(Resolution(2048, 1080));
+	resolutions.push_back(Resolution(1920, 1200));
+	resolutions.push_back(Resolution(1920, 1080));
+	resolutions.push_back(Resolution(1680, 1050));
+	resolutions.push_back(Resolution(1600, 1200));
+	resolutions.push_back(Resolution(1600,  900));
+	resolutions.push_back(Resolution(1440,  900));
+	resolutions.push_back(Resolution(1400, 1050));
+	resolutions.push_back(Resolution(1280, 1024));
+	resolutions.push_back(Resolution(1280,  800));
+	resolutions.push_back(Resolution(1280,  720));
+	resolutions.push_back(Resolution(1152,  864));
+	resolutions.push_back(Resolution(1024,  768));
+	resolutions.push_back(Resolution(800 ,  600));
+	resolutions.push_back(Resolution(640 ,  480));
+	resolutions.push_back(Resolution(320 ,  240));
+	resolutions.push_back(Resolution(320 ,  200));
+	return resolutions;
+}
+
+static std::vector<Resolution> resolutions = initResolutions();
```

Comment:
Good idea moving this to the graphics code. :)

But: using a global std::vector<>, filled by the initialization code like this isn't really all that great. We want to avoid doing that if possible. I'd rather you use a plain static array of Resolution structs. That can be filled out directly, too, and can live in the data section of the binary. (I probably should have done that in the first place :) )

Also, we don't need this data to be mutable, right? So it should be const. Also, also, please name it something like "kResolutions".

See kLanguageStrings in src/aurora/language_strings.h (https://github.com/xoreos/xoreos/blob/master/src/aurora/language_strings.h#L41) for how it should probably look like. You can then use the ARRAYSIZE macro to iterate over this array (see https://github.com/xoreos/xoreos/blob/master/src/aurora/language.cpp#L258).

Another nitpick: It should be "Refactor", not "Refaktor" in the commit message. :P